lib/orm/models.js

```diff
@@ -28,39 +28,88 @@ function createAssociationOptionsForFields(fields) {
       };
 
-      associationOptions[fieldName] = isList
-        ? hasMany(associationName, options)
-        : belongsTo(associationName, options);
+      associations.push({
+        name: associationName,
+        fieldName: fieldName,
+        type: isList ? "hasMany" : "belongsTo",
+        options,
+      });
     }
 
-    return associationOptions;
+    return associations;
   };
 }
 
-function createAssociationOptions(type) {
+function createAssociations(type) {
   const fields = type.getFields();
-  const associationOptions = Object.keys(fields).reduce(
-    createAssociationOptionsForFields(fields),
-    {}
-  );
+  return Object.keys(fields).reduce(createAssociationForFields(fields), []);
+}
 
-  return associationOptions;
+function createModel(type) {
+  return {
+    name: type.name,
+    camelizedName: camelize(type.name),
+    associations: createAssociations(type),
+  };
 }
 
-function ensureModel(mirageSchema, type) {
-  const associationOptions = createAssociationOptions(type);
-  const model = Model.extend(associationOptions);
+function shouldCreateModel(type) {
+  return (
+    isObjectType(type) && !isRelayType(type) && !type.name.startsWith("__")
+  );
+}
 
-  mirageSchema.registerModel(type.name, model);
+function shouldRegisterModel(mirageSchema, name) {
+  return !mirageSchema.hasModelForModelName(name);
 }
 
-function shouldAddModel(mirageSchema, type) {
-  return (
-    !mirageSchema.hasModelForModelName(type.name) &&
-    isObjectType(type) &&
-    !isRelayType(type) &&
-    !type.name.startsWith("__")
+function registerModel(mirageSchema, model) {
+  const associationsOptions = model.associations.reduce(
+    (options, association) => {
+      options[association.fieldName] =
+        association.type === "hasMany"
+          ? hasMany(association.name, association.options)
+          : belongsTo(association.name, association.options);
+
+      return options;
+    },
+    {}
   );
+
+  mirageSchema.registerModel(model.name, Model.extend(associationsOptions));
+}
+
+/**
+ * Create a POJO for all model definitions from the Graphql Schema. It figures
+ * out all the models and their relationships that need to be established for
+ * the Mirage Schema.
+ *
+ * Use this data to generate Mirage Models. This is the underlining
+ * implementation of `ensureModels`.
+ *
+ * @function createModels
+ * @param {{graphQLSchema: Object} options
+ */
+export function createModels({ graphQLSchema }) {
+  const models = [];
+
+  const graphQLSchemaQueryTypes = [
+    graphQLSchema.getMutationType(),
+    graphQLSchema.getQueryType(),
+    graphQLSchema.getSubscriptionType(),
+  ];
+  const typeMap = graphQLSchema.getTypeMap();
+
+  Object.keys(typeMap).forEach(function (typeName) {
+    const { type } = unwrapType(typeMap[typeName]);
+    const isQueryType = graphQLSchemaQueryTypes.includes(type);
+
+    if (shouldCreateModel(type) && !isQueryType) {
+      models.push(createModel(type));
+    }
+  });
+
+  return models;
```

Comment:
Very small comment here...

Why not use `reduce` here and assign the result to `models` vs. the `forEach` styles. I know this is super minor and I apologize in advance.